test/http.cc

```diff
@@ -231,6 +234,56 @@ TEST_P(HttpTest, GetHeaders) {
   auto response = future.get();
 
   EXPECT_EQ(200, response.code());
-  EXPECT_THAT(response.headers(), testing::ContainsRegex("Foo: Bar"));
+  EXPECT_THAT(
+      response.headers(),
+      testing::Contains(http::Header("Foo", "Bar")));
+  EXPECT_EQ(response.headers().at("Content-Length"), "25");
+  EXPECT_EQ("<html>Hello World!</html>", response.body());
+}
+
+
+TEST_P(HttpTest, GetDuplicateHeaders) {
+  std::string scheme = GetParam();
+
+  HttpMockServer server(scheme);
+
+  // NOTE: using an 'http::Client' configured to work for the server.
+  http::Client client = server.Client();
+
+  EXPECT_CALL(server, ReceivedHeaders)
+      .WillOnce([](auto socket, const std::string& data) {
+        EXPECT_THAT(data, testing::ContainsRegex("foo: bar1, bar2"));
+
+        socket->Send(
+            "HTTP/1.1 200 OK\r\n"
+            "Foo: Bar1\r\n"
+            "Foo: Bar2\r\n"
```

Comment:
Based on the RFC that you commented on above, doesn't this violate the rule? Clearly curl handles it given this test passes, but is that what you're trying to test? Or were you just trying to test multiple header values?

Reply:
From what I read I understand that server "can" do this and we, as a client, "must not" do this.
Take a look: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie